apps/emqx/src/emqx_broker_mon.erl

```diff
@@ -0,0 +1,211 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2024 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+-module(emqx_broker_mon).
+
+-feature(maybe_expr, enable).
+
+-behaviour(gen_server).
+
+%% API
+-export([
+    start_link/0,
+
+    get_mnesia_tm_mailbox_size/0,
+    get_broker_pool_max_mailbox_size/0
+]).
+
+%% `gen_server' API
+-export([
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2
+]).
+
+%%------------------------------------------------------------------------------
+%% Type declarations
+%%------------------------------------------------------------------------------
+
+-define(PT_KEY, {?MODULE, atomics}).
+-define(mnesia_tm_mailbox, mnesia_tm_mailbox).
+-define(broker_pool_max_mailbox, broker_pool_max_mailbox).
+-define(GAUGES, #{
+    ?mnesia_tm_mailbox => 1,
+    ?broker_pool_max_mailbox => 2
+}).
+
+-ifdef(TEST).
+-define(UPDATE_INTERVAL_MS, persistent_term:get({?MODULE, update_interval}, 10_000)).
+-else.
+-define(UPDATE_INTERVAL_MS, 10_000).
+-endif.
+
+-define(update, update).
+-define(mnesia_tm_high_watermark, 50).
```

Comment:
`high_watermark` -> `threshold`
we have some existing configs named high watermark for legacy reasons, i personally would like to have them renamed (with old as alias) to threshold too (as a separate task to follow up though) wdyt?.

Reply:
renamed the new configs to use "threshold"
:checkered_flag: 

the old ones might make sense to be called watermark, if they are lower and upper bounds?  even if we do want to rename them, I'd say that it would be best to change that in a separate PR.